test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -226,6 +226,72 @@ test.describe( 'Cover', () => {
 		await expect( overlay ).toHaveCSS( 'background-color', 'rgb(0, 0, 0)' );
 		await expect( overlay ).toHaveCSS( 'opacity', '0.5' );
 	} );
+
+	test( 'z-index of Navigation block inside a Cover block is higher than the other cover inner blocks', async ( {
```

Comment:
> On the topic of tests, given the complexities around when the cover block should and shouldn't be visible due to z-index styling, it would be great if we could have some tests to protect against future regressions.

@aaronrobertshaw, this was a bit challenging. Basically, I couldn't find a way to check if an element would be over the other. It could be a screenshot, but it seems it's an approach we are not using in Gutenberg tests and it could easily change.

But I created a helper function (`calculateZIndexContext`) to get the zIndex of the context where the element is. It's also not perfect because if we have an element parent of both blocks with a `z-index`, both would have the same because we're not comparing in what level of the tree they are. Since we aren't testing it with other blocks as wrapper, it would be fine unless at some point we decide to add some wrappers to the editor with `z-index`, but I suspect we should be fine here.

Reply:
Would `toBeVisible` be sufficient here? I'm not sure what the main goal of this test is. I wonder if we could test this from the accessibility perspective. If a modal is opened, then all the other elements behind it should be inaccessible and [hidden](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-hidden). I don't think checking the `z-index` is good either since it's an implementation detail; having a higher `z-index` doesn't mean it's top in the stacking context.

Reply:
Thanks for weighing in @kevin940726 🙇 

> Would toBeVisible be sufficient here? 

My understanding of the [Playwright docs](https://playwright.dev/docs/actionability#visible) are that `toBeVisible` or `toBeHidden` won't quiet be enough as they don't take z-index into account at all.

> Element is considered visible when it has non-empty bounding box and does not have visibility:hidden computed style.
> 
> Note that according to this definition:
> 
> Elements of zero size are not considered visible.
> Elements with display:none are not considered visible.
> Elements with opacity:0 are considered visible.

When I initially suggested a test here I was hoping we could get away with something like `toBeVisible` to avoid testing the implementation details. Given that isn't straightforward, I don't think we want this test to block a fix landing. 

Sorry to have sent you on a wild goose chase @renatho 🙏 

> If a modal is opened, then all the other elements behind it should be inaccessible and [hidden](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-hidden).

There's also another issue where certain elements/blocks behind the navigation overlay can still be interacted with.  Could that throw another spanner in the works?

That said, it does spark another possible idea. Maybe we could do a `.click( { trial: true } )` on the elements in question. It should return true if the element can be clicked and false if it's overlapped?  As the nav overlay should fill the page and completely cover the hidden block, I'm thinking we wouldn't need to worry about a position for the click.

Take that with a grain of salt. It's definitely an odd suggestion.

Reply:
> Maybe we could do a [.click( { trial: true } )](https://playwright.dev/docs/api/class-locator#locator-click-option-trial) on the elements in question.

Oh! That's a great suggestion! That was what I was trying initially and I didn't find. Updated here: https://github.com/WordPress/gutenberg/pull/66093/commits/88944b99366a419f7d1eff1fca17ded0eafcce18

The only cons is that it needs to wait until the timeout, but I forced it to be 1 second. I think that's fine. WDYT?

Reply:
I'm not sure. Ultimately, I like having a test to guard against regressions here. So to me if that is the cost of having that protection I could live with it but I don't have strong opinions. I'd like to defer the final call there to others that have more knowledge around Playwright etc.

I'll resolve this thread as the inline comment for the code in question will catch the final decision.